src/System.Net.Sockets/tests/FunctionalTests/UnixDomainSocketTest.cs

```diff
@@ -202,6 +202,67 @@ public void Socket_SendReceive_Success()
         }
 
         [OuterLoop] // TODO: Issue #11345
+        [Theory]
+        [InlineData(5000, 1, 1, false)]
+        [InlineData(5000, 1, 1, true)]
+        public async Task Socket_SendReceiveAsync_PropagateToStream_Success(int iterations, int writeBufferSize, int readBufferSize, bool shutdownUnusedDirections)
+        {             
+            var writeBuffer = new byte[writeBufferSize * iterations];
+            new Random().NextBytes(writeBuffer);
+            var readData = new MemoryStream();
+
+            string path = GetRandomNonExistingFilePath();
+            var endPoint = new UnixDomainSocketEndPoint(path);
+            try
+            {
+                using (var server = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified))
+                using (var client = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified))
+                {
+                    server.Bind(endPoint);
+                    server.Listen(1);
+
+                    Task clientReceives = Task.Run(async () =>
+                    {
+                        await client.ConnectAsync(endPoint);
+
+                        int bytesRead;
+                        byte[] buffer = new byte[readBufferSize];
+                        while ((bytesRead = await client.ReceiveAsync(new ArraySegment<byte>(buffer), SocketFlags.None)) > 0)
+                        {
+                            readData.Write(buffer, 0, bytesRead);
+                        }
+                    });
+
+                    using (Socket accepted = await server.AcceptAsync())
+                    {
+                        if (shutdownUnusedDirections)
+                        {
+                            server.Shutdown(SocketShutdown.Receive);
+                            client.Shutdown(SocketShutdown.Send);
```

Comment:
Yeah, I refactored some stuff around and this no longer makes sense here.  Will fix.